### PR TITLE
MAINT: Updated tick and category test formatting

### PR DESCRIPTION
**Pull request by madphysicist**
Rearranged tests in `lib/matplotlib/tests/test_ticker.py` and made a couple of very minor changes to `lib/matplotlib/tests/test_category.py`. Based on comment by @story645 in PRs #7482 and #7965.

This PR should probably be merged before either of the others so I can rebase them onto it and clean things up.

### CODE REVIEW

File: lib/matplotlib/tests/test_ticker.py [hunk 1]
```diff
@@ -482,82 +500,102 @@ def test_LogFormatterSciNotation():
...
+    @pytest.mark.parametrize(
+            'xmax, decimals, symbol, x, display_range, expected',
+            percent_data, ids=percent_ids)
+    def test_percentformatter(self, xmax, decimals, symbol,
+                              x, display_range, expected):
+        formatter = mticker.PercentFormatter(xmax, decimals, symbol)
+        assert formatter.format_pct(x, display_range) == expected
+
+
+class TestEndFormatter(object):
```
Comment (story645) [inline lib/matplotlib/tests/test_ticker.py]: typo: EngFormatter

File: lib/matplotlib/tests/test_category.py [hunk 2]
```diff
@@ -122,6 +122,25 @@ def lt(tl):
...
+    ]
+
+    bytes_ids = ['string list', 'bytes list', 'bytes ndarray']
+
+    numlike_data = [
+        ['1', '11', '3'],
+        np.array(['1', '11', '3']),
+        [b'1', b'11', b'3'],
+        np.array([b'1', b'11', b'3'])],
+    ]
```
Comment (story645) [inline lib/matplotlib/tests/test_category.py]: apparently an extra one here messing up appveyor?

**Comment by story645** [comment]
😄 thanks!

**Comment by madphysicist** [comment]
Thanks for looking through the PRs. I fixed up the issues you found.

**Comment by madphysicist** [comment]
I am pretty sure that this PR is ready to go. The failing tests appear to be caused by an external source.

**Comment by madphysicist** [comment]
No more failing tests.

**Comment by madphysicist** [comment]
@dstansby. I've been seeing [MRG+1] a lot recently on PR titles. What does it mean?

**Comment by story645** [comment]
It means your PR has been approved by one reviewer (matplotlib asks for 2 approvals before commiting). http://matplotlib.org/devel/coding_guide.html#pr-review-guidelines

**Comment by madphysicist** [comment]
Can you be the other one?

File: lib/matplotlib/tests/test_ticker.py [hunk 3]
```diff
@@ -482,82 +501,104 @@ def test_LogFormatterSciNotation():
...
+
+class TestEngFormatter(object):
+    def test_formatting(self):
+        """
+        Create two instances of EngFormatter with default parameters, with and
+        without a unit string ('s' for seconds). Test the formatting in some
+        cases, especially the case when no SI prefix is present, for values in
+        [1, 1000).
+
+        Should not raise exceptions.
```
Comment (story645) [inline lib/matplotlib/tests/test_ticker.py]: don't need this comment.

File: lib/matplotlib/tests/test_ticker.py [hunk 4]
```diff
@@ -482,82 +501,104 @@ def test_LogFormatterSciNotation():
...
+        """
+        unitless = mticker.EngFormatter()
```
Comment (story645) [inline lib/matplotlib/tests/test_ticker.py]: these should probably be two separate tests (unitless and with unit) and possibly parameterized

**Comment by story645** [comment]
Sorry thought I approved. Have two comments, but if you wanna punt those to a different improve engineering test PR, let me know and I'll merge.

**Comment by madphysicist** [comment]
@story645 I made the improvements here. I kept is as a single parametrized test, removed the extra comment and also took care of some extra whitespace in ticker.py.

**Comment by madphysicist** [comment]
Not an actual failure, Mac test just didn't run.

**Comment by QuLogic** [comment]
With #7973 merged, this has conflicts. When you rebase, please be sure to leave out any empty `cleanup` decorators.

**Comment by story645** [comment]
I think on the rebase or something, you accidentally introduced a conflict. Please resolve.

**Comment by madphysicist** [comment]
@QuLogic Should I replace `@cleanup(style='classic')` with `@pytest.mark.style('classic')`, and make all similar changes that make sense?

**Comment by QuLogic** [comment]
Classic is the default now, it's not necessary if that's really what's being used (and I realize I forgot to remove a couple.)

**Comment by madphysicist** [comment]
In that case I will remove those too.

**Comment by madphysicist** [comment]
Just to make triple sure, I should make the replacement for `'default'` though?

**Comment by QuLogic** [comment]
That's correct.

**Comment by madphysicist** [comment]
Done. I ended up parametrizing a couple of the tests and adding one `pytest.mark.style('default')` that I don't think was there before to make things pass. Should be good now.

**Comment by madphysicist** [comment]
Again looks like Mac test didn't run. No actual failures.

**Comment by madphysicist** [comment]
I've closed and reopened the PR to restart Travis and hopefully get the Mac test to run, but if you approve, this PR is ready to go.

File: lib/matplotlib/tests/test_ticker.py [hunk 5]
```diff
@@ -12,267 +12,282 @@
...
+
+
+class TestMultipleLocator(object):
+    def test_basic(self):
+        loc = mticker.MultipleLocator(base=3.147)
+        test_value = np.array([-9.441, -6.294, -3.147, 0., 3.147, 6.294,
+                               9.441, 12.588])
+        assert_almost_equal(loc.tick_values(-7, 10), test_value)
+
+    def test_MultipleLocator_set_params(self):
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_ticker.py]: Seems like you forgot to remove the class in this one.

File: lib/matplotlib/tests/test_ticker.py [hunk 6]
```diff
@@ -12,267 +12,282 @@
...
+        ax.xaxis.set_minor_locator(mticker.LogLocator(base=10,
+                                                      subs=np.arange(2, 10)))
+        ax.xaxis.set_major_formatter(mticker.LogFormatter(labelOnlyBase=True))
+        ax.xaxis.set_minor_formatter(mticker.LogFormatter(labelOnlyBase=False))
+        # axis range above 3 decades, only bases are labeled
+        ax.set_xlim(1, 1e4)
+        fmt = ax.xaxis.get_major_formatter()
+        fmt.set_locs(ax.xaxis.get_majorticklocs())
+        show_major_labels = [fmt(x) != '' for x in
+                                            ax.xaxis.get_majorticklocs()]
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_ticker.py]: Not aligned properly; better to wrap at `for`.

File: lib/matplotlib/tests/test_ticker.py [hunk 7]
```diff
@@ -296,59 +310,60 @@ def get_view_interval(self):
...
-
-    for base in test_cases:
+    ]
+
+    base_data = [2.0, 5.0, 10.0, np.pi, np.e]
+
+    @pytest.mark.parametrize(
+            'labelOnlyBase, exponent, locs, positions, expected', param_data)
+    @pytest.mark.parametrize('base', base_data)
+    def test_LogFormatterExponent(self, labelOnlyBase, base, exponent,
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_ticker.py]: Class name is still here.

File: lib/matplotlib/tests/test_ticker.py [hunk 8]
```diff
@@ -296,59 +310,60 @@ def get_view_interval(self):
...
+
+    def test_blank(self):
+        # Should be a blank string for non-integer powers if labelOnlyBase=True
+        formatter = mticker.LogFormatterExponent(base=10, labelOnlyBase=True)
+        formatter.axis = FakeAxis()
+        assert formatter(10**0.1) == ''
+
+
+class TestLogFormatterSciNotation(object):
+    testdata = [
```
Comment (QuLogic) [inline lib/matplotlib/tests/test_ticker.py]: Why `testdata` and not `test_data` like the others?

Comment (madphysicist) [inline lib/matplotlib/tests/test_ticker.py]: I wasn't sure if it would cause any problems if I started it with `test_`.

**Comment by madphysicist** [comment]
@QuLogic Made the following changes:
- Removed 2 class names in method names.
- Renamed `testdata` to `test_data`.
- Wrapped list comprehension with `for` on new line.

File: lib/matplotlib/tests/test_ticker.py (hunk 8 above)
Comment (QuLogic) [inline lib/matplotlib/tests/test_ticker.py]: We'll see how the tests go, then, but I'd guess this shouldn't be a problem since it isn't callable.

**Comment by madphysicist** [comment]
Ready to go?

**Pull request merged**